### PR TITLE
feat(worker): cds/git/env context as environment variable in plugin

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19479.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19480.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19480.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Review by richardlt — APPROVED** [review]

File: engine/worker/internal/runV2.go [hunk 1]
```diff
@@ -425,3 +430,45 @@ func (w *CurrentWorker) computeContextForAction(ctx context.Context, parentConte
...
+
+	}
+
+	var mapGIT map[string]interface{}
+	btsGIT, _ := json.Marshal(contexts.Git)
+	if err := json.Unmarshal(btsGIT, &mapGIT); err != nil {
+		return nil, sdk.NewErrorFrom(sdk.ErrInvalidData, "unable to unmarshal git context")
+	}
+	for k, v := range mapGIT {
+		newEnvVar[fmt.Sprintf("GIT_%s", strings.ToUpper(k))] = fmt.Sprintf("%q", v)
```
Comment (yesnault) [inline engine/worker/internal/runV2.go:459]:
```suggestion
		newEnvVar[fmt.Sprintf("GIT_%s", strings.ToUpper(k))] = fmt.Sprintf("%v", v)
```

File: engine/worker/internal/runV2.go [hunk 2]
```diff
@@ -425,3 +430,45 @@ func (w *CurrentWorker) computeContextForAction(ctx context.Context, parentConte
...
+	if err := json.Unmarshal(btsCDS, &mapCDS); err != nil {
+		return nil, sdk.NewErrorFrom(sdk.ErrInvalidData, "unable to unmarshal cds context: %v", err)
+	}
+	for k, v := range mapCDS {
+		switch reflect.ValueOf(v).Kind() {
+		case reflect.Map, reflect.Slice:
+			s, _ := json.Marshal(v)
+			newEnvVar[fmt.Sprintf("CDS_%s", strings.ToUpper(k))] = string(s)
+		default:
+			newEnvVar[fmt.Sprintf("CDS_%s", strings.ToUpper(k))] = fmt.Sprintf("%q", v)
```
Comment (yesnault) [inline engine/worker/internal/runV2.go:448]:
```suggestion
			newEnvVar[fmt.Sprintf("CDS_%s", strings.ToUpper(k))] = fmt.Sprintf("%v", v)
```

File: tests/08_v2_analyze.yml [hunk 3]
```diff
@@ -108,6 +108,7 @@ testcases:
     - script: "cat MyFirstWorkflow-1-0-myFirstJob-workflowEnv | grep \"workflowValue\""
     - script: "cat MyFirstWorkflow-1-0-myFirstJob-stepEnv | grep \"stepValue\""
     - script: "cat MyFirstWorkflow-1-0-myThirdJob-jobEnv | grep \"jobValue\""
+    - script: "cat MyFirstWorkflow-1-0-alwaysJob-mystep | grep \"alwaysJob\""
```
Comment (yesnault) [inline tests/08_v2_analyze.yml:111]:
```suggestion
    - script: "cat MyFirstWorkflow-1-0-alwaysJob-mystep | grep \"I am alwaysJob\""
```

File: engine/worker/internal/runV2.go [hunk 4]
```diff
@@ -425,3 +430,45 @@ func (w *CurrentWorker) computeContextForAction(ctx context.Context, parentConte
...
+	var mapEnv map[string]interface{}
+	btsEnv, _ := json.Marshal(contexts.Env)
+	if err := json.Unmarshal(btsEnv, &mapEnv); err != nil {
+		return nil, sdk.NewErrorFrom(sdk.ErrInvalidData, "unable to unmarshal env context")
+	}
+	for k, v := range mapEnv {
+		if strings.HasPrefix(k, "CDS_") || strings.HasPrefix(k, "GIT_") {
+			continue
+		}
+		newEnvVar[strings.ToUpper(k)] = fmt.Sprintf("%q", v)
```
Comment (yesnault) [inline engine/worker/internal/runV2.go:471]:
```suggestion
		newEnvVar[strings.ToUpper(k)] = fmt.Sprintf("%v", v)
```

**Review by yesnault — CHANGES_REQUESTED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-cds#19558.0 ✘  
* Build
  * Build and Package All ✘  
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-cds#19559.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19559.0 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

**Review by yesnault — APPROVED** [review]

**Pull request merged**